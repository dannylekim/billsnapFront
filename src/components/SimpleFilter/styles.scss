@import "../../variables";

div.quick__sorting__container {
    position: absolute;
    top: 75px;
    left: 275px;
    height: 120px;

    ul.sorting__content {
      width: 103px;
      transform: translateX(470px);
      border-radius: 6px;
      height: inherit;
      background-color: $light-gray-background;
      list-style: none;
      padding-top: 10px;
      padding-left: 0;

      li.sorting__titles {
        font-family: $primary-font;
        font-weight: 500;
        font-size: 20px;
        line-height: 26px;
        color: $secondary-text-color;
      }

      li.sorting__titles:hover {
        background-color: #bebebe;
      }
    }
  }