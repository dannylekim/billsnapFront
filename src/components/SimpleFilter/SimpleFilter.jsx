import React from "react";
import PropType from "prop-types";

import "./styles.scss";

export const SHORT_FILTER_OPTIONS = ["Newest", "A to Z", "Z to A", "Oldest"];

const SimpleFilter = ({
    applyFilter,
    currentActive
}) => {
  return (
    <div className='quick__sorting__container'>
      <ul className='sorting__content'>
        { SHORT_FILTER_OPTIONS
          .filter((title) => title !== currentActive)
          .map((title, key) => (
            <li
              className='sorting__titles'
              onClick={() => applyFilter(title)}
              key={key}>
              {title}
            </li>
          ))}
      </ul>
    </div>
  );
};

SimpleFilter.propTypes = {
    applyFilter: PropType.func.isRequired,
    currentActive: PropType.string
};

export default SimpleFilter;