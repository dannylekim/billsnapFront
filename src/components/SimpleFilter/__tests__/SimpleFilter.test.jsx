import React from "react";
import SimpleFilter from "../SimpleFilter";

describe("SimpleFilter", () => {
  describe("render", () => {
    let handleMockFunction;
    let mockSortingType;
    
    beforeEach(() => {
        handleMockFunction = jest.fn();
        mockSortingType = "";
    });

    describe("snapshots ðŸ“¸", () => {
      it("SimpleFilter should match snap shot without sorting type", () => {
        matches(
            <SimpleFilter
                applyFilter={handleMockFunction}
                currentActive={mockSortingType}
            />
        );
      });
      it("SimpleFilter should match snap shot when sorting type is set", () => {
        mockSortingType = 'A to Z';

        matches(
            <SimpleFilter
                applyFilter={handleMockFunction}
                currentActive={mockSortingType}
            />
        );
      });
    });

    describe("interactions", () => {
      it("should call prop handler on click", () => {
        const wrapper = shallow(<SimpleFilter
            applyFilter={handleMockFunction}
            currentActive={mockSortingType}
        />);

        wrapper.find(".sorting__titles").at(1).simulate('click');

        expect(handleMockFunction).toHaveBeenCalled();
      });
    });
  });
});
