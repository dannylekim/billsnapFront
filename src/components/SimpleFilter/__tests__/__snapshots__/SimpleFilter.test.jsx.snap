// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleFilter render interactions SearchBar should match snap shot when sorting type is set 1`] = `
<div
  className="quick__sorting__container"
>
  <ul
    className="sorting__content"
  >
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Newest
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Z to A
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Oldest
    </li>
  </ul>
</div>
`;

exports[`SimpleFilter render interactions SearchBar should match snap shot without sorting type 1`] = `
<div
  className="quick__sorting__container"
>
  <ul
    className="sorting__content"
  >
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Newest
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      A to Z
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Z to A
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Oldest
    </li>
  </ul>
</div>
`;

exports[`SimpleFilter render interactions should call prop handler on click 1`] = `
<div
  className="quick__sorting__container"
>
  <ul
    className="sorting__content"
  >
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Newest
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Z to A
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Oldest
    </li>
  </ul>
</div>
`;

exports[`SimpleFilter render interactions should match snap shot when sorting type is set 1`] = `
<div
  className="quick__sorting__container"
>
  <ul
    className="sorting__content"
  >
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Newest
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Z to A
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Oldest
    </li>
  </ul>
</div>
`;

exports[`SimpleFilter render snapshots ðŸ“¸ SearchBar should match snap shot when sorting type is set 1`] = `
<div
  className="quick__sorting__container"
>
  <ul
    className="sorting__content"
  >
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Newest
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Z to A
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Oldest
    </li>
  </ul>
</div>
`;

exports[`SimpleFilter render snapshots ðŸ“¸ SearchBar should match snap shot without sorting type 1`] = `
<div
  className="quick__sorting__container"
>
  <ul
    className="sorting__content"
  >
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Newest
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      A to Z
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Z to A
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Oldest
    </li>
  </ul>
</div>
`;

exports[`SimpleFilter render snapshots ðŸ“¸ SimpleFilter should match snap shot when sorting type is set 1`] = `
<div
  className="quick__sorting__container"
>
  <ul
    className="sorting__content"
  >
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Newest
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Z to A
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Oldest
    </li>
  </ul>
</div>
`;

exports[`SimpleFilter render snapshots ðŸ“¸ SimpleFilter should match snap shot without sorting type 1`] = `
<div
  className="quick__sorting__container"
>
  <ul
    className="sorting__content"
  >
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Newest
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      A to Z
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Z to A
    </li>
    <li
      className="sorting__titles"
      onClick={[Function]}
    >
      Oldest
    </li>
  </ul>
</div>
`;
